include/flatbuffers/flatbuffers.h

```diff
@@ -1182,11 +1182,16 @@ class Verifier FLATBUFFERS_FINAL_CLASS {
     #ifdef FLATBUFFERS_DEBUG_VERIFICATION_FAILURE
       assert(ok);
     #endif
+    if (!ok)
+        upper_bound_ = 0;
     return ok;
   }
 
   // Verify any range within the buffer.
   bool Verify(const void *elem, size_t elem_len) const {
+    const uint8_t * const uppper_bound = reinterpret_cast<const uint8_t *>(elem) + elem_len;
```

Comment:
Done. 
Though I think adding a number is close to 0 when it comes to speed penalties.